### PR TITLE
[15.0][MIG] sale_order_restrict_cancel_existing_invoice

### PR DESCRIPTION
**Pull request by Saran440**
Migrated code support with version 15

### CODE REVIEW

**Comment by rousseldenis** [comment]
/ocabot migration sale_order_restrict_cancel_existing_invoice

File: sale_order_restrict_cancel_existing_invoice/models/sale_order.py [hunk 1]
```diff
@@ -20,14 +19,13 @@ def _get_confirmed_invoice_states(self):
...
-        return ["open"]
+        return ["posted"]
 
-    @api.multi
     @api.depends("invoice_ids")
     def _compute_confirmed_invoice_ids(self):
         for sale in self:
             sale.confirmed_invoice_ids = sale.invoice_ids.filtered(
-                lambda i, states=self._get_confirmed_invoice_states(): i.state in states
+                lambda l: l.state in self._get_confirmed_invoice_states()
```
Comment (rousseldenis) [inline sale_order_restrict_cancel_existing_invoice/models/sale_order.py]: @Saran440 At least change to 'move' as it is not a line 'l'

**Review by rousseldenis — APPROVED** [review]
Code review. Minor comment

File: sale_order_restrict_cancel_existing_invoice/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def _get_confirmed_invoice_states(self):
+        """
+        Return the invoice states that should avoid order cancel
+        :return:
+        """
+        return ["posted"]
+
+    @api.depends("invoice_ids")
+    def _compute_confirmed_invoice_ids(self):
+        for sale in self:
```
Comment (rousseldenis) [inline sale_order_restrict_cancel_existing_invoice/models/sale_order.py:27]:
```suggestion
        states = self._get_confirmed_invoice_states()
        for sale in self:
            sale.confirmed_invoice_ids = sale.invoice_ids.filtered(
                lambda invoice, states=states: invoice.state in states
```
In fact, my original code was wrong. :sweat_smile:

Comment (rousseldenis) [reply sale_order_restrict_cancel_existing_invoice/models/sale_order.py:27]: @Saran440

Comment (Saran440) [reply sale_order_restrict_cancel_existing_invoice/models/sale_order.py:27]: @rousseldenis Fixed. Thank you for help :)

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**